experimental/Schemes/duValSing.jl

```diff
@@ -0,0 +1,513 @@
+    set_attribute!(I2,:is_absolutely_prime,true)
+    ## pass to algebraic extension
+    r_changed = base_ring(I2)
+    kk = coefficient_ring(r_changed)
+    J_changed = ideal(r_changed,  [change_coefficient_ring(kk,a, parent = r_changed) for a=gens(J)])
+    tempvec = decide_du_val_singularity(Spec(quo(r_changed,J_changed)[1]),I2)
+    for x in tempvec
+      x[1] || return [x]
+      x = (x[1],x[2],x[3],mult)
+      push!(result_vector,x)
+    end
+  end
+
+  return result_vector
+end
+
+@doc raw"""
+    _check_du_val_at_point(IX:Ideal,Ipt::Ideal))
+
+Returns a tuple T with the following data:
+- T[1]::Bool has V(IX) at most a du Val singularity at V(Ipt)
+- T[2]::Tuple Type of du Val singularity at V(Ipt)
+
+**Note**: Assumes Ipt to be absolutely irreducible.
+
+**Note**: Do not call directly, only via the higher level functions is_du_Val_singularity and decide_du_Val_singularity.
+
+"""
+function _check_duval_at_point(IX::Ideal,Ipt::Ideal)
+  R = base_ring(IX)
+  R == base_ring(Ipt) || error("basering mismatch")
+  kk = base_ring(R)
+  characteristic(kk) == 0 || error("only available in characteristic zero")
+ 
+  JM = jacobi_matrix(gens(IX))
+
+  smooth = (:A,0)
+
+  ## localize at point
+  a = Oscar.rational_point_coordinates(Ipt)
+  U = complement_of_point_ideal(R,a)
+  RL, loc_map = localization(R,U)
+  IX_loc = loc_map(IX)
+  JM_loc =  map(x ->loc_map(x), JM[:,:])
+
+  if !all(iszero(a))
+    F_loc = free_module(RL,ngens(IX))
+    Jm = sub(F_loc,JM_loc)[1]
+    Jm = Jm + (loc_map(IX)*F_loc)[1]
+    Jm_shifted = Oscar. shifted_module(Jm)[1]
+    F_shifted = ambient_free_module(Jm_shifted)
+  else
+    F = free_module(R,ngens(IX))
+    Jm = sub(F,JM)[1]
+    Jm_shifted = Jm + (IX * F)[1]
+    F_shifted = F
+  end
+
+  o = negdegrevlex(R)*lex(F_shifted)
+  F1 = leading_module(Jm_shifted,o)
+  F1quo = quo(F_shifted, F1)[1]
+
+  constant_mons = vector_space_dimension(F1quo,0) 
+  constant_mons < 2 || return (false, typeof(smooth))                   ## not a hypersurface
+  constant_mons > 0 || return (true, smooth)                            ## no singularity
+  
+  tau = vector_space_dimension(F1quo)
+
+  corank = vector_space_dimension(F1quo,1)
+  corank < 3 || return (false,typeof(smooth))                           ## at least T_3,3,3 not duVal
+  corank > 1 || return (true,(:A,tau))                                  ## A_series
+
+  # we now already know essentially a hypersurface of corank 2, count degrees of freedom cubicpart
+  cubiccount = vector_space_dimension(F1quo,2)
+  cubiccount < 3 || return  (false, typof(smooth))                      ## at least X_9
+  cubiccount > 1 || return  (true, (:D,tau))                            ## D_series
+
+  # it is definitely in the E/J series
+  tau < 9 || return(false, typeof(smooth))                              ## at least J_10
+  return (true, (:E,tau))                                               ## E_6, E_7, E_8
+end
+
+@doc raw"""
+    vector_space_dimension(M::SubquoModule, d::Int)
```

Comment:
Please move this to
src/Modules/Ungraded_Modules

Notifying @wdecker @fieker @jankoboehm , that we have some new module functionality here.